site/app/libraries/Utils.php

```diff
@@ -240,32 +240,42 @@ public static function getCompareByReference() {
      * students_version is an array of user and their highest submitted version
      */
 
-    public static function getAutoFillData($students, $students_version = null){
+    public static function getAutoFillData($students, $students_version = null) {
         $students_full = array();
         $null_section = array();
         foreach ($students as $student) {
             $student_entry = array('value' => $student->getId(),
                     'label' => $student->getDisplayedFirstName() . ' ' . $student->getDisplayedLastName() . ' <' . $student->getId() . '>');
-            $students_full[] = $student_entry;
-            if($students_version != null){
-                if($student->getRegistrationSection() != null && array_key_exists($student->getId(),$students_version)){
+            if($students_version != null) {
+                if($student->getRegistrationSection() != null && array_key_exists($student->getId(),$students_version)) {
                     if ($students_version[$student->getId()] !== 0) {
                         $student_entry['label'] .= ' (' .
                         $students_version[$student->getId()] . ' Prev Submission)';
                     }
                 }
-            }else{
+                $students_full[] = $student_entry;
+            } else {
                 $null_entry = array('value' => $student->getId(),
-                'label' => '[NULL section] ' . $student->getDisplayedFirstName() . ' ' . $student->getDisplayedLastName() . ' <' . $student->getId() . '>'); 
+                'label' => '[NULL section]' . $student->getDisplayedFirstName() . ' ' . $student->getDisplayedLastName() . ' <' . $student->getId() . '>'); 
 
                 $in_null_section = false;
                 foreach ($null_section as $null_student) {
                     if($null_student['value'] === $student->getId()) $in_null_section = true;
                 }
-                if(!$in_null_section) $null_section[] = $null_entry;
+                if(!$in_null_section) $null_section[] = $null_entry; 
+                //$students_full = self::removeStudentWithId($students_full, 'value', $student->getId());
             }
         }
         $students_full = array_unique(array_merge($students_full, $null_section), SORT_REGULAR);
         return json_encode($students_full);
     }
+
+    public static function removeStudentWithId($students, $key, $id) {
```

Comment:
Please do not add code into the codebase if you are not going to use it. For reasons, [this](https://stackoverflow.com/questions/15699995/why-unused-code-should-be-deleted) could be a good reference.

Reply:
I ended up using the function, since adding the else-if is not regarded as an ideal solution.

Reply:
I ended up using the function, since adding the else-if is not regarded as an ideal solution.